manifests/fcontext.pp

```diff
@@ -63,50 +63,42 @@
 # }
 #
 define selinux::fcontext (
-  $destination = '',
+  $pathname,
+  $destination = undef,
   $context     = '',
-  $pathname    = '',
   $filetype    = false,
-  $filemode    = '',
+  $filemode    = undef,
   $equals      = false,
-  $policy      = 'targeted',
 ) {
-  Exec {
-    path => '/bin:/sbin:/usr/bin:/usr/sbin',
-  }
-  if $pathname == '' {
-    fail('pathname must not be empty')
-  }
 
-  if ( $equals == false ) and ( $pathname == '' ) {
-    fail('pathname must not be empty')
+  if !defined(Class['selinux']) {
+    fail('You must include the selinux base class before using any selinux defined resources')
```

Comment:
That is a possibility, but I've always preferred making it explicit what modules were going to do.  Many of the resources required selinux::package which would also fail with an undefined resource, this just makes it a bit more explicit.